### ISSUE TITLE
CallstackDeminifier requires SourceCodeProvider?

### ISSUE THREAD

**Issue by habibialkaff** [Microsoft/sourcemap-toolkit#51]
Hi,

This may not be a good place to ask this, but perhaps someone can give some answers.
The SourceMapConsumer in https://github.com/mozilla/source-map#sourcemapconsumer only requires the sourceMap content. So why does the GetStackTraceDeminfier require ISourceCodeProvider? Shouldn't souceMap be enough to deminify a stack trace?

Thanks

**Comment by christiang88** [comment]
This is a fine place to ask. It's a great question.

So when you get a stack trace in the browser, it gives you the line number where the exception was thrown. 

Something like this:
    at i (http://localhost:11323/crashcauser.min.js:1:113)

In order to accurately get the original method name corresponding to the minified method i, we need the line number of i. However, the stack trace gives you the line number where the exception happened, not the line number of the function declaration. We use the original source code to be able to find line number of the function declaration that contains the line number of the exception. In the example, above, it would tell us the line and column number of the declaration of i. Once we have the location information of i, we can look in the source map for the original name.

So to summarize, we don't need the source code to deminify the method name, but we need it to get the location of the function wrapping the exception in the stack trace.

**Comment by christiang88** [comment]
Also, to clarify, you don't need the source code to parse the source map, you just need it if you want to deminify a stack trace accurately.

**Comment by habibialkaff** [comment]
Thanks for the information. This is really interesting. Not just the original line and column number of the stack trace, but It can also get the original name of an anonymized function.

I'll read the source code further to get a better understanding on how it works.

I'll close this issue. Thank you very much
